src/python/pants/base/workunit.py

```diff
@@ -109,6 +109,8 @@ def __init__(self, run_info_dir, parent, name, labels=None, cmd='', log_config=N
     self.parent = parent
     self.children = []
 
+    self.zipkin_attrs = None
```

Comment:
Would it be possible to require these as a parameter to the constructor? Or is it actually necessary for the `zipkin_attrs` to be set _after_ the workunit has been constructed?

The advantage to passing them to the constructor would be that you could ensure that they were never `None`, which reduces error cases.

Reply:
Updated the code and added comments